maze_transformer/evaluation/plot_maze.py

```diff
@@ -31,44 +15,193 @@ class PathFormat:
     (path_true, "true", "-", "red")
     """
 
-    path: NDArray
+    path: CoordArray
     label: str | None = None
     fmt: str | None = None
     color: str | None = None
+    line_width: float | None = None
     quiver_kwargs: dict | None = (None,)
 
 
-def plot_multi_paths(
-    maze: LatticeMaze,
-    paths: list[PathFormat | tuple | list],
-    show: bool = True,
-) -> None:
-    # show the maze
-    plt.imshow(
-        maze.as_img(),
-        cmap="gray",
-        extent=[
-            -0.75,
-            maze.grid_shape[0] - 1 + 0.75,
-            maze.grid_shape[1] - 1 + 0.75,
-            -0.75,
-        ],
-    )
-
-    # plot paths
-    for pf in paths:
-        if isinstance(pf, (tuple, list)):
-            pf = PathFormat(*pf)
-
-        # p_transformed: NDArray = maze.points_transform_to_img(pf.path)
-        p_transformed: NDArray = np.array(pf.path)
+class MazePlot:
+    """Class for displaying mazes and paths
+
+    UNIT_LENGTH: Set ratio between node size and wall thickness in image.
+    Wall thickness is fixed to 1px
+    A "unit" consists of a single node and the right and lower connection/wall.
+    Example: ul = 14 yields 13:1 ratio between node size and wall thickness
+    """
+
+    UNIT_LENGTH = 14
+    DEFAULT_PREDICTED_PATH_COLORS = [
+        "tab:blue",
+        "tab:green",
+        "tab:purple",
+        "tab:orange",
+        "tab:olive",
+        "tab:cyan",
+    ]
+
+    def __init__(self, maze: LatticeMaze) -> None:
+        self.maze: LatticeMaze = maze
+        self.true_path: PathFormat = None
+        self.predicted_paths: list = []
+
+    def add_true_path(
+        self,
+        path: CoordArray,
+        label: str = "true path",
+        fmt: str = "--",
+        color: str = "red",
+        line_width: float = 2.5,
+        quiver_kwargs: dict | None = None,
+    ) -> MazePlot:
+        """
+        Recieve true path and formatting preferences from input and save as dict in true_path variable.
+        """
+        self.true_path = PathFormat(
+            path=path,
+            label=label,
+            fmt=fmt,
+            color=color,
+            line_width=line_width,
+            quiver_kwargs=quiver_kwargs,
+        )
+        return self
+
+    def add_predicted_path(
+        self,
+        path: CoordArray,
+        label: str | None = None,
+        fmt: str = ":",
+        color: str | None = None,
+        line_width: float = 2,
+        quiver_kwargs: dict | None = {"width": 0.015},
+    ) -> MazePlot:
+        """
+        Recieve predicted path and formatting preferences from input and save in predicted_path list.
+        Default formatting depends on nuber of paths already saved in predicted path list.
+        """
+        if label is None:
+            label = "predicted path {path_num}".format(
+                path_num=len(self.predicted_paths) + 1
+            )
+        if color is None:
+            color = self.DEFAULT_PREDICTED_PATH_COLORS[len(self.predicted_paths)]
```

Comment:
this will throw if len(predicted_paths) > len(DEFAULT_PREDICtED_PATH_COLOrS)

We should either:
* Fix with a modulo operation or `itertools.cycle`
* Raise a custom exception

Reply:
added modulo as it requires minimal changes